example/Makefile

```diff
@@ -1,26 +1,32 @@
-.PHONY: default run build clean
-
-SRCS := $(wildcard ../*.go)
+SHELL := bash
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
 
+.PHONY: default
 default: run
 
-x.wasm: main.go $(SRCS)
+srcs := $(wildcard ../**/*.go)
+
+x.wasm: main.go $(srcs)
 	@echo "Building WASM module..."
-	@tinygo build -o x.wasm -target=wasi main.go
+	@tinygo build -o $@ -target=wasi main.go
```

Comment:
styleguide suggested using the automatic vars (in this case `$@`, target name) when possible. I wasn't totally sure how to specify the first dependency `main.go`, as it was suggested we do that as well.

Reply:
Pretty sure you would use `$<` (similar to how you did in the top-level makefile) to specify `main.go` here.
```makefile
x.wasm: main.go $(srcs)
	@echo "Building WASM module..."
	@tinygo build -o $@ -target=wasi $<
```

The `$<` automatic variable represents the first "prerequisite" for the target, which in this case is `main.go`

Reply:
updated as you suggested, thanks!